base/expr.jl

```diff
@@ -5,7 +5,7 @@ symbol(s::ASCIIString) = symbol(s.data)
 symbol(s::UTF8String) = symbol(s.data)
 symbol(a::Array{UInt8,1}) =
     ccall(:jl_symbol_n, Any, (Ptr{UInt8}, Int32), a, length(a))::Symbol
-symbol(x::Char) = symbol(string(x))
+symbol(s...) = symbol(print_to_bytearray(s...))
```

Comment:
This should probably be 
```julia
symbol(s...) = symbol(string(s...))
```
Because `string` contains specializations for common operations, (eg: two `ASCIIString`s). With the string rework, the performance difference is likely to be smaller than today.